File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int a, int b) {
    int answer = 0;
    
    if(a % 2 && b % 2) {
        answer = (a * a) + (b * b);
    }else if(a % 2 || b % 2) {
        answer = 2 * (a + b);
    }else {
        answer = a > b ? a - b : b - a;
    }
    
    
    return answer;
}

File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.py
def solution(a, b):
    answer = 0
    
    # 모두 홀수, 하나 홀수, 모두 아님
    if a % 2 and b % 2:
        answer = a**2 + b**2
    elif a % 2 or b % 2:
        answer = 2 * (a + b)
    else:
        answer = abs(a - b)
    
    return answer